### ISSUE TITLE
Option to download original trace / fuzzer stdout/err output

### ISSUE THREAD

**Issue by sobukus** [google/oss-fuzz#2634]
I cannot find a way to just download the log file of a fuzzer run (the original stacktrace). In the case of https://oss-fuzz.com/testcase-detail/5067888303538176 , there is very bogus behavour leading to the final UBSan crash. 20k lines is not too much to handle for a file on disk with UNIX tools, but it is unbearable in the heavily scripted website. Uncollapsing the log makes Firefox ask me if I want to stop the website scripts because things are stalling (on a Core i3 laptop with 16G RAM) and it takes 22 seconds. Just too heavy.

It is a bit ironic that the website contains lots of scripting (20k lines in itself, coincidentally) to be able to quickly load only the relevant part of a log file / stack trace. Just downloading the whole log would be quicker in this case.

Anyhow: Is there a way to simply download the file? Even after waiting for the browser to load it all, it is too cumbersome to navigate and I want my shell tools to work on the trace.

**Comment by inferno-chromium** [comment]
You can download the fuzzing logs by visiting the fuzzer stats page here for UBSan
https://oss-fuzz.com/fuzzer-stats/by-fuzzer/date-start/2019-07-23/date-end/2019-07-29/fuzzer/libFuzzer/job/libfuzzer_ubsan_mpg123

For this crash, i pasted the log in bug. It is important to fix fuzz target to not log this much output, so please improve fuzz target code in https://github.com/google/oss-fuzz/tree/master/projects/mpg123 to disable noisy/unneeded log message. website loads fine for me in Chrome, can you please try it there.
Also, note that this is an unreproducible crash, so can be ignored if you can't make a hang of it. This just tells that crash only happens in an hour of fuzzing but not with any individual testcase. So, it could be a bug in fuzz target itself saving state somehow.

**Comment by sobukus** [comment]
Thanks for the hint, but I cannot find the log from the report there. I get pointed to a Google Cloud bucket with logs from July 2019. The testcase was produced on Tue, Mar 26, 2019, 12:34 AM, according to https://oss-fuzz.com/testcase-detail/5067888303538176. That seems rather early to me, but I don't know which delays in reporting are normal for OSS-Fuzz.

Anyhow, how would I locate the exact log file for this test run, if for some reason it is contained in the folder for July? I got a timestamp in the distant past, I got a long number identifying the run … I cannot match that to a log file name that I can locate:-/

I don't really see this issue as closed if it is normal that I see the report long after the log file is gone from the bucket. Or isn't it? So is the answer that I need to scrape it from the report page after all? The data is still there, just not very accessible.

To answer your hint about excessive output: The extreme logging of the fuzz target is a direct result of the corruption behind the (bogus?) error. Also, the output is valuable in this case, as it demonstrates early that something is wrong: The file offset reported for invalid MPEG headers jumps around wildly, while it should increase monotonically. The alternative is no output at all from libmpg123. Well, if the testcase is reproducable, this is no problem … maybe I can whip up a pull request for that.

It may be that Chrome is more efficient with the scripting around the log file display, your machine being faster than my laptop, so that it maybe takes only 2 seconds for you. My point is that a direct link for plain text download would be infinitely more efficient as I wouldn't use a browser to render it at all. When we silence the fuzzers using MPG123_QUIET, the output is reduced to the few lines of the sanitizer info and copy'n'paste is an option again. Just not the best.
